c10/core/TensorImpl.h

```diff
@@ -1447,10 +1420,34 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
    * or strides.
    */
   void refresh_contiguous() {
-    TORCH_INTERNAL_ASSERT(!is_variable());  // TODO: remove this when Variable and Tensor are merged
     is_contiguous_ = compute_contiguous();
   }
 
+  /**
+   * Copy the storage pointer and the tensor metadata fields (e.g. sizes / strides / storage_offset)
+   * from one TensorImpl to another TensorImpl.
+   *
+   * For usage of `version_counter` and `allow_tensor_metadata_change`, see NOTE [ TensorImpl Shallow-Copying ].
+   */
+  static void copy_tensor_data(
```

Comment:
Did it not work to make it protected?

Reply:
It is protected now, with the keyword at https://github.com/pytorch/pytorch/pull/17072/files#diff-7f54d357d87885923a1536316f17ac6bR1410.